src/components/WalletInit/CreateWallet/MnemonicCheckScreen.js

```diff
@@ -114,24 +102,21 @@ const MnemonicCheckScreen = () => {
 
 export default MnemonicCheckScreen
 
-const MnemonicInput = ({
-  partialPhrase,
-  error,
-  onPress,
-}: {
-  partialPhrase: Array<string>,
+type MnemonicInputProps = {
+  userEntries: Array<Entry>,
   error: boolean,
-  onPress: (word: string) => any,
-}) => {
+  onPress: (Entry) => any,
+}
+const MnemonicInput = ({userEntries, error, onPress}: MnemonicInputProps) => {
   return (
     <View style={styles.recoveryPhrase}>
       <View style={[styles.recoveryPhraseOutline, error && styles.recoveryPhraseError]}>
-        {partialPhrase.map((word, index, array) => {
+        {userEntries.map((entry, index, array) => {
           const isLast = index === array.length - 1
 
           return (
-            <View key={word} style={[styles.wordBadgeContainer, !isLast && styles.selected]}>
-              <WordBadge word={word} disabled={!isLast} onPress={isLast ? () => onPress(word) : undefined} />
+            <View key={entry.id} style={[styles.wordBadgeContainer, !isLast && styles.selected]}>
+              <WordBadge word={`${entry.word} x`} disabled={!isLast} onPress={() => onPress(entry)} />
```

Comment:
I moved the x from word badge, so only the mnemonic input displays the x  and not the word badge list

Reply:
> it might make more sense to have a single "backspace" button, instead of an "x" on each badge, then it's easier to remove multiple entries

@wolverineks , agree

Reply:
I've noticed the removed `x` in the word badge now. Works for now then, but can be improved with a backspace button in the future, yes